test/unit/core/type_traits/range_iterator_test.cpp

```diff
@@ -53,11 +53,18 @@ TEST(range_and_iterator, sentinel)
 template <typename list1, typename list2, size_t pos = 0>
 void expect_same_types()
 {
-    constexpr bool val = std::is_same_v<seqan3::list_traits::at<pos, list1>, seqan3::list_traits::at<pos, list2>>;
+    using type_1 = seqan3::list_traits::at<pos, list1>;
+    using type_2 = seqan3::list_traits::at<pos, list2>;
+
+    // Make sure types such as `long` and `long long` are treated as equal if they are equivalent (macOS).
+    constexpr bool val = std::is_same_v<type_1, type_2> ||
+        (std::unsigned_integral<type_1> && std::unsigned_integral<type_2> && sizeof(type_1) == sizeof(type_2)) ||
+        (std::signed_integral<type_1> && std::signed_integral<type_2> && sizeof(type_1) == sizeof(type_2));
+
     if constexpr (!val)
     {
-        std::cerr << "pos: " << pos << " \'" << seqan3::detail::type_name_as_string<seqan3::list_traits::at<pos, list1>>
-                  << "\' not equal to \'" << seqan3::detail::type_name_as_string<seqan3::list_traits::at<pos, list2>>
+        std::cerr << "pos: " << pos << " \'" << seqan3::detail::type_name_as_string<type_1>
+                  << "\' not equal to \'" << seqan3::detail::type_name_as_string<type_2>
                   << "\' \n";
     }
     EXPECT_TRUE(val);
```

Comment:
I wouldn't want to change this here.

I would just change the offending test case

```patch
-------------- test/unit/core/type_traits/range_iterator_test.cpp --------------
index 008deca9c..417d6d69d 100644
@@ -158,15 +158,21 @@ TEST(range_and_iterator, difference_type)
 {
     //TODO(h-2): add something that actually has a different difference_type
     using iterator_of_int_vector = std::ranges::iterator_t<std::vector<int>>;
     using foreign_iterator = seqan3::detail::random_access_iterator<std::vector<int>>;
-    auto v = std::views::iota(1);
     using type_list_example = seqan3::type_list<std::ranges::range_difference_t<std::vector<int>>, // short
                                                 typename std::vector<int>::difference_type, // member type
                                                 std::ranges::range_difference_t<std::vector<int> const>, // const container
                                                 std::iter_difference_t<iterator_of_int_vector>, // iterator
-                                                std::iter_difference_t<foreign_iterator>, // iterator2
-                                                std::ranges::range_difference_t<decltype(v)>>; // range, no member
+                                                std::iter_difference_t<foreign_iterator>>; // range, no member
+
+    using comp_list = seqan3::type_list<std::ptrdiff_t,
+                                        std::ptrdiff_t,
+                                        std::ptrdiff_t,
+                                        std::ptrdiff_t,
+                                        std::ptrdiff_t>;
+
+    expect_same_types<type_list_example, comp_list>();
 
     // views::ints' difference_type is not std::ptrdiff_t, but depends on the size.
     // For infinite views, like in our case, it's std::int_fast64_t (or std::int_fast32_t on 32bit).
     // On most platforms this is the same as std::size_t (`long int` on Linux and FreeBSD;
@@ -176,16 +182,13 @@ TEST(range_and_iterator, difference_type)
     using view_int_diff_t = std::conditional_t<sizeof(std::ptrdiff_t) == 8,
                                                std::int_fast64_t,
                                                std::int_fast32_t>;
 
-    using comp_list = seqan3::type_list<std::ptrdiff_t,
-                                        std::ptrdiff_t,
-                                        std::ptrdiff_t,
-                                        std::ptrdiff_t,
-                                        std::ptrdiff_t,
-                                        view_int_diff_t>;
-
-    expect_same_types<type_list_example, comp_list>();
+    auto v = std::views::iota(1);
+    using iota_difference_t = std::ranges::range_difference_t<decltype(v)>;
+    EXPECT_TRUE((std::signed_integral<iota_difference_t>));
+    EXPECT_EQ(sizeof(iota_difference_t), sizeof(view_int_diff_t));
 }
 
 TEST(range_and_iterator, size_type)
 {
```